spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/MeanAbsolutePercentageCriterion.scala

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2016 The BigDL Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.intel.analytics.bigdl.nn
+
+import breeze.linalg.sum
+import com.intel.analytics.bigdl.nn.abstractnn.TensorCriterion
+import com.intel.analytics.bigdl.tensor.{DenseTensorApply, Tensor, TensorFunc6}
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
+
+import scala.reflect.ClassTag
+
+/**
+ * This method is same as `mean_absolute_percentage_error` loss in keras.
+ * It caculates diff = K.abs((x - y) / K.clip(K.abs(x), K.epsilon(), Double.MaxValue))
+ * and return 100 * K.mean(diff) as outpout
+ * Here, the x and y can have or not have a batch.
+ * @param ev$1
+ * @param ev
+ * @tparam T The numeric type in the criterion, usually which are [[Float]] or [[Double]]
+ */
+class MeanAbsolutePercentageCriterion[T: ClassTag]
+  (implicit ev: TensorNumeric[T]) extends TensorCriterion[T] {
+
+  @transient
+  private var buffer1 : Tensor[T] = null
```

Comment:
private var buffer1 : Tensor[T] = Tensor[T]() ?
and we can avoid the == null check in `updateOutput`

Reply:
buffer1 is a transient variable. When doing criterion clone (serialization clone), buffer1 will be null, so " == null check" can not be deleted.